cherab/solps/solps_plasma.py

```diff
@@ -248,55 +597,201 @@ def eirene_simulation(self):
         :rtype: Eirene
         """
         if self._eirene is None:
-            raise RuntimeError("EIRENE simulation data not available for this SOLPS simulation")
+            raise RuntimeError("EIRENE simulation data not available for this SOLPS simulation.")
         else:
             return self._eirene
 
+    @eirene_simulation.setter
+    def eirene_simulation(self, value):
+        if not isinstance(value, Eirene):
+            raise ValueError('Attribute "eirene_simulation" must be an Eirene instance.')
+
+        self._eirene = value
+
+    def b2_flux_to_velocity(self, poloidal_flux, radial_flux, parallel_velocity):
```

Comment:
Wouldn't it be better to make this method a standalone instead of class method, like it is in the case of EFITEquilibrium class? This class is becoming quite big, it would help decreasing it. Also the method itself could be useful/used for various purposes. The same could be done with `eirene_flux_to_velocity`.